src/Ledger/RolloverLedger/RolloverLedgerContainer.js

```diff
@@ -151,19 +155,54 @@ export const RolloverLedgerContainer = ({ resources, mutator, match, history, lo
     return initValues;
   }, [ledger, budgets, toFiscalYearId, toFiscalYearSeries, series, funds, currentFiscalYear]);
 
-  const callRollover = useCallback(() => {
+  const checkIsRolloverAlreadyExist = useCallback(async (currentRollover) => {
+    try {
+      const { fromFiscalYearId: fromFY, toFiscalYearId: toFY } = currentRollover;
+      const query = `ledgerId==${ledgerId} and rolloverType=="${LEDGER_ROLLOVER_TYPES.commit}" and fromFiscalYearId==${fromFY} and toFiscalYearId==${toFY}`;
+      const { totalRecords } = await ky.get(LEDGER_ROLLOVER_API, { searchParams: { query } }).json();
+
+      if (totalRecords) {
+        const { name: ledgerName } = ledger || {};
+        const fromFYcode = fiscalYears?.find(({ id }) => id === fromFY)?.code;
+        const toFYcode = fiscalYears?.find(({ id }) => id === toFY)?.code;
+
+        const message = intl.formatMessage(
+          {
+            id: 'ui-finance.ledger.rollover.error.conflict',
+          },
+          {
+            ledgerName,
+            fromFYcode,
+            toFYcode,
+          },
+        );
+
+        showCallout({
+          message,
+          type: 'error',
+        });
+
+        return true;
+      }
+
+      return false;
+    } catch (error) {
+      throw new Error(error);
+    }
+  }, [fiscalYears, intl, ky, ledger, ledgerId, showCallout]);
+
+  const callRollover = useCallback(async () => {
     const rolloverCb = ifRolloverPreview(savingValues) ? testRollover : rollover;
+    const isRolloverExist = await checkIsRolloverAlreadyExist(savingValues);
```

Comment:
It seems that the user needs to take a few extra steps to get an error about an existing rollover - checking for unpaid invoices and confirming the operation. Perhaps we should move the rollover check to the beginning of the operation, in the `showConfirmation` cb? What do you think?

Reply:
@NikitaSedyx what is your opinion about this comment?

Reply:
Is it better if we use the Loader component since the whole process may take 2-5 seconds?